src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -472,6 +472,39 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
         matrixType,
         MatrixValue(matrixType, globals, colValues, rvd)))
 
+  private type Axis = Int
+  private val globalAxis: Axis = 0
+  private val rowAxis: Axis = 1
+  private val colAxis: Axis = 2
+  private val entryAxis: Axis = 3
+
+  private def useIR(axis: Axis, ast: AST): Boolean = {
+    if (hc.forceIR)
+      return true
+    axis match {
+      case this.globalAxis =>
+        ast.`type`.asInstanceOf[TStruct].size < 500
+
+      case this.rowAxis =>
+        ast.`type`.asInstanceOf[TStruct].size < 500 &&
+          globalType.size < 3 &&
+          colType.size == 1 &&
+          Set(TInt32(), +TInt32(), TString(), +TString())
+            .contains(colType.types(0))
```

Comment:
what's the reasoning for this?

Reply:
I think this covers a lot of common cases where you've got either just a string col ID or you've created something from an index/range. Mostly idk, I didn't want it to just check that there was only one field but have that field be something huge, but I don't have super strong feels about keeping this.